### PR TITLE
Changed img-cmd script to work with new Oldenborg dataset

### PR DESCRIPTION
**Pull request by chauvuha**

### CODE REVIEW

File: img_cmd_training_new_dataset.py [hunk 1]
```diff
@@ -0,0 +1,280 @@
+from argparse import ArgumentParser
+
+import matplotlib.pyplot as plt
+import os.path
```
Comment (anthonyjclark) [inline img_cmd_training_new_dataset.py]: We probably want to use `pathlib` instead of `os.path`

File: img_cmd_training_new_dataset.py [hunk 2]
```diff
@@ -0,0 +1,280 @@
...
+
+
+dataset_path = artifact_dir + '/'   # Path to the extracted images from the artifact
+
+from torch.utils.data import Dataset
+
+# Constants 
+# Load the dataset artifact
+
+VALID_PCT = 0.05
```
Comment (anthonyjclark) [inline img_cmd_training_new_dataset.py]: These should probably all be program arguments.

File: img_cmd_training_new_dataset.py [hunk 3]
```diff
@@ -0,0 +1,280 @@
...
+MODEL_PATH_REL_TO_DATASET = Path("cmd_models_fixed")
+DATA_PATH_REL_TO_DATASET = Path("cmd_data_fixed")
+# VALID_MAZE_DIR = Path("../Mazes/validation_mazes8x8/")
+
+compared_models = {
+    "resnet18": resnet18,
+    "resnet34": resnet34
+}
+
+DATASET_DIR.ls()
```
Comment (anthonyjclark) [inline img_cmd_training_new_dataset.py]: Do you want to print this here? Or more likely just remove the line in this script since it is no longer a notebook.

File: img_cmd_training_new_dataset.py [hunk 4]
```diff
@@ -0,0 +1,280 @@
...
+
+def filename_to_class(filename: str) -> str:
```
Comment (anthonyjclark) [inline img_cmd_training_new_dataset.py:44]: This reminds me: in `boxnav` we should always give the angle and not set it to zero since we handle that here.

File: img_cmd_training_new_dataset.py [hunk 5]
```diff
@@ -0,0 +1,280 @@
...
+    angle = float(filename.split("_")[1].split(".")[0].replace("p", "."))
+    if angle > 0:
```
Comment (anthonyjclark) [inline img_cmd_training_new_dataset.py:46]: Instead of `> 0` we should come up with a threshold.

File: img_cmd_training_new_dataset.py [hunk 6]
```diff
@@ -0,0 +1,280 @@
...
+        :param index: (int) number that represents the location of the desired data
+        :return: (tuple) tuple of all the information associated with the desired data
+        """
+        # The filename of the image given a specific index
+        img_filename = self.all_filenames[index]            
+        
+        # Opens image file and ensures dimension of channels included
+        img = Image.open(img_filename).convert('RGB')
+        # Resizes the image
+        img = img.resize((224, 224))
```
Comment (anthonyjclark) [inline img_cmd_training_new_dataset.py]: `224` should probably be a parameter of the constructor.

File: img_cmd_training_new_dataset.py [hunk 7]
```diff
@@ -0,0 +1,280 @@
...
+        # If the index is not 0, the cmd is determined by the previous img_filename
+        if index != 0:
+            prev_img_filename = self.all_filenames[index-1]
+            cmd_name = filename_to_class(str(prev_img_filename))            
+        cmd = self.class_indices[cmd_name]
+        
+        # Data and the label associated with that data
+        return (img, cmd), label
+
+class cmd_model(nn.Module):
```
Comment (anthonyjclark) [inline img_cmd_training_new_dataset.py]:
```suggestion
class CommandModel(nn.Module):
```

File: img_cmd_training_new_dataset.py [hunk 8]
```diff
@@ -0,0 +1,280 @@
...
+    def __init__(self, arch: str, pretrained: bool):
+        super(cmd_model, self).__init__()
+        self.cnn = arch(pretrained=pretrained)
+        
+        self.fc1 = nn.Linear(self.cnn.fc.out_features + 1, 512)
```
Comment (anthonyjclark) [inline img_cmd_training_new_dataset.py:115]: We probably want to make this part of the network less arbitrary. Make a note that we should try other architectures here.

File: img_cmd_training_new_dataset.py [hunk 9]
```diff
@@ -0,0 +1,280 @@
...
+        self.r1 = nn.ReLU(inplace=True)
+        self.fc2 = nn.Linear(512, 3)
+        
+    def forward(self, data):
```
Comment (anthonyjclark) [inline img_cmd_training_new_dataset.py:119]: This method needs some documentation so that we don’t forget what it is doing on each line.

**Review by anthonyjclark — CHANGES_REQUESTED** [review]

File: img_cmd_training_new_dataset.py (hunk 5 above)
Comment (chauvuha) [reply img_cmd_training_new_dataset.py:46]: We are not quite sure we understand what you mean by threshold with the angle. Would you mind elaborating?

Comment (anthonyjclark) [reply img_cmd_training_new_dataset.py:46]: This requires two changes:

* In `boxnav` we should use the angle in the filenames without any thresholding.
* Here we should say that any angle between, for example, -5 degrees and 5 degrees is given the label `forward.

**Comment by anthonyjclark** [comment]
Work was completed elsewhere.
